src/evernodeone.c

```diff
@@ -48,264 +48,321 @@ int64_t hook(uint32_t reserved)
         uint32_t memo_len, type_len, format_len, data_len;
         GET_MEMO(0, memos, memos_len, memo_ptr, memo_len, type_ptr, type_len, format_ptr, format_len, data_ptr, data_len);
 
-        if (op_type == OP_HOST_REG)
+        if (op_type == OP_HOST_REG || op_type == OP_SET_HOOK)
         {
-            int is_valid = 0;
-            BUFFER_EQUAL_STR_GUARD(is_valid, format_ptr, format_len, FORMAT_TEXT, 1);
-            if (!is_valid)
-                rollback(SBUF("Evernode: Memo format should be text."), 1);
+            if (op_type == OP_HOST_REG)
```

Comment:
Do we need two if conditions? Wouldn't if else if enough wince there are no common code
```
if (op_type == OP_HOST_REG)
else if (op_type == OP_HOST_REG)
```